### DISCUSSION TITLE
Does `jax.experimental.shard_map.shard_map(..., auto=...)` implement xla auto-sharding?

### DISCUSSION THREAD

**Discussion by man2machine** [jax-ml/jax#24736] [Q&A]
Does the `auto` argument call the experimental XLA auto-sharding algorithm in https://github.com/openxla/xla/tree/main/xla/hlo/experimental/auto_sharding which is based on the Alpa paper https://github.com/alpa-projects/alpa

If so, when auto-sharding is performed, does it perform only shard-parallelism (intra-operator) or pipeline-parallelism (inter-operator) as well as described in the Alpa paper?

**Comment by dcbdan** [comment]
I have the same question. Maybe [this](https://github.com/jax-ml/jax/pull/19738) pull request is enabling Alpa.
